src/app/attributeformconfig/qgsattributeformcontaineredit.h

```diff
@@ -0,0 +1,45 @@
+/***************************************************************************
+    qgsattributeformcontaineredit.h
+    ---------------------
+    begin                : February 2020
+    copyright            : (C) 2020 Denis Rouzaud
+    email                : denis@opengis.ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSATTRIBUTEFORMCONTAINEREDIT_H
+#define QGSATTRIBUTEFORMCONTAINEREDIT_H
+
+#include <QWidget>
+
+#include "ui_qgsattributeformcontaineredit.h"
+
+#include "qgis_app.h"
+
+class QTreeWidgetItem;
+
+/**
+ * Widget to edit a container (tab or group box) of a form configuration
+ */
+class APP_EXPORT QgsAttributeFormContainerEdit: public QWidget, private Ui_QgsAttributeFormContainerEdit
+{
+    Q_OBJECT
+
+  public:
+    explicit QgsAttributeFormContainerEdit( QTreeWidgetItem *item, QWidget *parent = nullptr );
+
+
+    void updateItemData();
```

Comment:
Doesn't it need docs?

Reply:
It's in app, I added a short description for the class only